snuba/utils/streams/batching.py

```diff
@@ -247,14 +245,17 @@ def _flush(self, force: bool = False) -> None:
                 "batch.flush.normalized", flush_duration / batch_results_length
             )
 
-        logger.debug("Committing offsets")
+        logger.debug("Committing offsets for batch")
         commit_start = time.time()
-        self._commit()
+        self.consumer.stage_offsets(
+            {
+                partition: offsets.hi + 1
```

Comment:
This is kind of kludgy, alternatively the `Offsets` structure could represent the lower bound inclusive and upper bound exclusive (like slice syntax) which might make more sense anyway.

Reply:
What's the impact of changing the Offset structure ? (read: how many things do we expect to silently break :D ? )

Reply:
It is only used in `BatchingConsumer` to my knowledge (and its usage is "private" within that) so hopefully nothing.